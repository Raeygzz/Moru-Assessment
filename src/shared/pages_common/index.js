export * from "./dashboard/DashboardDetails";
export * from "./modal/editUserDetail/EditUserDetailModal";
