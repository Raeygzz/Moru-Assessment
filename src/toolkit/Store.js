import { configureStore, combineReducers } from "@reduxjs/toolkit";
import Logger from "redux-logger";

import UsersReducer from "./features/UsersSlice";

//combine reducer
const combinedReducer = combineReducers({
  users: UsersReducer,
});

// root reducer
const rootReducer = (state, action) => {
  // store reset
  if (action.type === "users/resetStore") {
    state = undefined;
  }

  return combinedReducer(state, action);
};

//store configure
const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }).concat(Logger),
});

export default store;
