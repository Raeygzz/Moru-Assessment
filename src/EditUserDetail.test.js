import { render, screen } from "@testing-library/react";

import { EditUserDetailModal } from "./shared/pages_common";

import { Provider } from "react-redux";
import Store from "./toolkit/Store";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders EditUserDetailModal correctly", () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <Provider store={Store}>
      <EditUserDetailModal />
    </Provider>
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
