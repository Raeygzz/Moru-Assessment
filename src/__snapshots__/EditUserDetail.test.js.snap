// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EditUserDetailModal correctly 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <EditUserDetailModal />
</Context.Provider>
`;
